src/Nethermind/Nethermind.Blockchain/Producers/TxPoolTxSource.cs

```diff
@@ -141,6 +142,8 @@ bool HasEnoughFounds(IDictionary<Address, UInt256> balances, Transaction transac
             int selectedTransactions = 0;
             int i = 0;
             
+            // TODO: removing transactions from TX pool here seems to be a bad practice since they will
```

Comment:
Unless we change the parent we want to build on (normally it should be the Head) I don't think its an issue. We might still remove those from here as after newest changes to TxPool this might have become obsolete